zephyr/drivers/can/can_sccan.c

```diff
@@ -0,0 +1,1082 @@
+		LOG_DBG("Filter added. ID: %d", filter_id);
+	}
+
+	if (need_enable) {
+		/* Enable CAN if disabled in this API */
+		sc_can_enable(config);
+	}
+
+unlock:
+	k_mutex_unlock(&data->inst_mutex);
+
+	return ret;
+}
+
+static void sc_can_remove_rx_filter(const struct device *dev, int filter_id)
+{
+	const struct sc_can_cfg *config = dev->config;
+	struct sc_can_data *data = dev->data;
+	int ret;
+	bool do_disable = false;
+
+	if (filter_id >= config->max_filter) {
+		LOG_ERR("Filter ID of %d exceeds maximum (%d)", filter_id, config->max_filter);
+		return;
+	}
+
+	k_mutex_lock(&data->inst_mutex, K_FOREVER);
+
+	if (sc_can_is_enabled(config)) {
+		/* Disable CAN with retry if enabled */
+		ret = sc_can_disable(config);
+		if (ret != 0) {
+			goto unlock;
+		}
+		do_disable = true;
+	}
+
+	sys_clear_bits(config->reg_addr + SCCAN_AFER_OFFSET, BIT(filter_id));
+
+	data->rx_cb_data_list[filter_id].rx_cb = NULL;
+
+	LOG_DBG("Filter removed. ID: %d", filter_id);
+
+	if (do_disable) {
+		/* Enable CAN if disabled in this API */
+		sc_can_enable(config);
+	}
+
+unlock:
+	k_mutex_unlock(&data->inst_mutex);
+
+	return;
+}
+
+static int sc_can_init(const struct device *dev)
+{
+	const struct sc_can_cfg *config = dev->config;
+	struct sc_can_data *data = dev->data;
+	struct can_timing timing;
+	int32_t ret;
+	uint32_t v;
+
+	data->tx_head = 0;
+	data->tx_tail = 0;
+
+	data->state = CAN_STATE_STOPPED;
+	data->state_change_cb = NULL;
+
+	/* Set timing according to dts default setting */
+	timing.sjw = config->sjw;
+	ret = can_calc_timing(dev, &timing, config->bus_speed, config->sample_point);
+	if (ret == -EINVAL) {
+		LOG_ERR("Can't find timing for given param");
+		return -EIO;
+	}
+	sc_can_set_timing(dev, &timing);
+
+	sc_can_enable_irq(dev);
+
+	/* Dump Version information */
+	v = sys_read32(config->reg_addr + SCCAN_VER_OFFSET);
+	LOG_DBG("Space Cubics CAN controller v%d.%d.%d initialized", SCCAN_VER_MAJOR(v),
+		SCCAN_VER_MINOR(v), SCCAN_VER_PATCH(v));
+
+	return 0;
+}
+
+static int sc_can_get_core_clock(const struct device *dev, uint32_t *rate)
+{
+	const struct sc_can_cfg *config = dev->config;
+
+	*rate = config->clock_frequency;
+
+	return 0;
+}
+
+/*
+ * This API specification is here:
+ *   Get the maximum standard (11-bit) CAN ID filters if false, or extended (29-bit)
+ *   CAN ID filters if true
+ * Acceptance filter of SC CAN Controller supports both standard and extended, so
+ * always returns the same value.
+ */
+static int sc_can_get_max_filters(const struct device *dev, bool ide)
+{
+	const struct sc_can_cfg *config = dev->config;
+
+	ARG_UNUSED(ide);
+
+	return config->max_filter;
+}
+
+static int sc_can_get_max_bitrate(const struct device *dev, uint32_t *max_bitrate)
+{
+	const struct sc_can_cfg *config = dev->config;
+
+	*max_bitrate = config->max_bitrate;
+
+	return 0;
+}
+
+static const struct can_driver_api sc_can_driver_api = {.get_capabilities = sc_can_get_capabilities,
+							.start = sc_can_start,
+							.stop = sc_can_stop,
```

Comment:
Would you please break after `{`?  So that it can be:

```c
static const struct can_driver_api sc_can_driver_api = {
	.get_capabilities = sc_can_get_capabilities,
	.start = sc_can_start,
	.stop = sc_can_stop,
	:
}

Reply:
It's result of clag-format with `zephyer/.clang-foamrt`.
These behavior was reported the other projects too.

i.g.
https://stackoverflow.com/questions/64061273/clang-format-how-to-format-c-struct-initializer-this-way

Maybe it's effect by `Cpp11BracedListStyle`, But current clang-format have so many issue related `Cpp11BracedListStyle`.

So we fix the format manually.